armory/data/datasets.py

```diff
@@ -680,12 +680,73 @@ def resisc45(
     )
 
 
+def check_shapes(actual, target):
+    """
+    Ensure that shapes match, ignoring None values
+
+    actual and target should be tuples
+        actual should not have None values
+    """
+    if type(actual) != tuple:
+        raise ValueError(f"actual shape {actual} is not a tuple")
+    if type(target) != tuple:
+        raise ValueError(f"target shape {target} is not a tuple")
+    if None in actual:
+        raise ValueError(f"None should not be in actual shape {actual}")
+    if len(actual) != len(target):
+        raise ValueError(f"len(actual) {len(actual)} != len(target) {len(target)}")
+    for a, t in zip(actual, target):
+        if a != t and t is not None:
+            raise ValueError(f"shape {actual} does not match shape {target}")
+
+
+class UCF101Context:
+    def __init__(self):
+        self.nb_classes = 101
+        self.input_type = np.uint8
+        self.x_shape = (None, 240, 320, 3)
+        self.frame_rate = 25
```

Comment:
It doesn't appear that this variable is used anywhere. Was it intended to be used in some sort of check, or just there for informative purposes?

The same applies to `self.nb_classes`.

Reply:
Primarily informational. Some models could use frame rate. I figure that this sort of object would make the most sense to store this sort of info.